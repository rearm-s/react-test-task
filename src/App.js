import React from "react";
import "bootstrap/dist/css/bootstrap.min.css";
import {Col, Container, Row} from "react-bootstrap";
import Header from "./components/Header";
import Home from "./components/Home";
import Employee from "./components/Employee";
import {Route, BrowserRouter as Router, Switch, Redirect} from "react-router-dom";
import {Provider} from "react-redux";
import store from "./redux/store";
import {Global, StyledWrapper} from "./StyledApp";


function App() {
    return <>
        <Global/>
        <Router>
            <StyledWrapper>
                <Provider store={store}>
                    <Container fluid style={{paddingLeft: 0, paddingRight: 0}}>
                        <Row>
                            <Col><Header/></Col>
                        </Row>
                        <Switch>
                            <Route exact path="/">
                                <Row>
                                    <Col><Home/></Col>
                                </Row>
                            </Route>
                            <Route path="/employee/:name">
                                <Row>
                                    <Col><Employee/></Col>
                                </Row>
                            </Route>
                            <Redirect to={"/"} />
                        </Switch>
                    </Container>
                </Provider>
            </StyledWrapper>
        </Router>
    </>
}

export default App;
