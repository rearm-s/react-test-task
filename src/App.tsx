import React from 'react';
import { Col, Container, Row } from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css';
import { Provider } from 'react-redux';
import { Route, BrowserRouter as Router, Switch, Redirect } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';

import { Employee, Header, Home } from './components';

import { StyledContainer, StyledWrapper, theme } from './styled/_Global';
import store from './redux/store';

const App = () => (
    <ThemeProvider theme={theme}>
        <StyledWrapper>
            <Provider store={store}>
                <Router>
                    <Container fluid as={StyledContainer}>
                        <Row>
                            <Col><Header/></Col>
                        </Row>
                        <Switch>
                            <Route exact path="/">
                                <Row>
                                    <Col><Home/></Col>
                                </Row>
                            </Route>
                            <Route path="/employee/:name">
                                <Row>
                                    <Col><Employee/></Col>
                                </Row>
                            </Route>
                            <Redirect to={'/'}/>
                        </Switch>
                    </Container>
                </Router>
            </Provider>
        </StyledWrapper>
    </ThemeProvider>
);

export default App;
