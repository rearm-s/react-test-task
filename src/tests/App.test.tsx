import { shallow } from 'enzyme';
import App from '../App';

describe('App', () => {
    it('should take snapshot', () => {
        const component = shallow(<App/>);
        expect(component).toMatchSnapshot();
    });
});
