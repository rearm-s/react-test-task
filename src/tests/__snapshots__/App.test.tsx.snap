// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should take snapshot 1`] = `ShallowWrapper {}`;
